### PR TITLE
boards: arm: add bl654_usb

### PR DESCRIPTION
**Pull request by kieran-mackey**
Adding board bl654_usb which is a Laird Connectivity BL654 module mounted on a
USB adapter.

Signed-off-by: Kieran Mackey <kieran.mackey@lairdconnect.com>

### CODE REVIEW

File: boards/arm/bl654_usb/doc/bl654_usb.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+#. Flash it onto the board. Note :file:`/dev/ttyACM0` is for Linux; it will be
+   something like ``COMx`` on Windows, and something else on macOS.
+
+   .. code-block:: console
+
+      nrfutil dfu usb-serial -pkg blinky.zip -p /dev/ttyACM0
+
+   When this command exits, observe the green LED on the board blinking,
+   instead of the red LED used by the bootloader.
```
Comment (lairdjm) [inline boards/arm/bl654_usb/doc/bl654_usb.rst:159]: No red or green LEDs

File: boards/arm/bl654_usb/Kconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	default y
+	depends on BOARD_BL654_USB
+
+config BOARD_HAS_NRF5_BOOTLOADER
+	bool "Board has nRF5 bootloader"
+	default y
+	help
+	  If selected, applications are linked so that they can be loaded by Nordic
+	  nRF5 bootloader.
+	depends on BOARD_BL654_USB
```
Comment (lairdjm) [inline boards/arm/bl654_usb/Kconfig:18]: Would remove this and always set the load offset to 0x1000, the MBR cannot be removed from this device so switching this off is an invalid build configuration

File: boards/arm/bl654_usb/bl654_usb.dts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		slot1_partition: partition@6e000 {
+			label = "image-1";
+			reg = <0x006e000 0x00005e000>;
+		};
+		storage_partition: partition@cc000 {
+			label = "storage";
+			reg = <0x000cc000 0x00004000>;
+		};
+		scratch_partition: partition@d0000 {
+			label = "image-scratch";
```
Comment (lairdjm) [inline boards/arm/bl654_usb/bl654_usb.dts:89]: Would double the storage size from 16KB to 32KB and reduce the scratch by 16KB from 64KB to 48KB as storage space is more useful

File: boards/arm/bl654_usb/doc/bl654_usb.rst (hunk 1 above)
Comment (greg-leach) [reply boards/arm/bl654_usb/doc/bl654_usb.rst:159]: Updated.

File: boards/arm/bl654_usb/bl654_usb.dts (hunk 3 above)
Comment (greg-leach) [reply boards/arm/bl654_usb/bl654_usb.dts:89]: Updated.

File: boards/arm/bl654_usb/Kconfig (hunk 2 above)
Comment (greg-leach) [reply boards/arm/bl654_usb/Kconfig:18]: Updated.

**Comment by cfriedt** [comment]
@MaureenHelm, @galak - could you please do a once over?

**Comment by thedjnK** [comment]
@MaureenHelm and @galak can we get this merged please?

**Review by carlescufi — APPROVED** [review]
This looks good to me as-is.

**Review by cfriedt — APPROVED** [review]
LGTM

**Pull request merged**